### PR TITLE
Added scroll progress option

### LINKED ISSUE #589: Expose scroll progress option
**Is your feature request related to a problem? Please describe.**
Currently the `Scroll` component allows users to trigger updates each time a waypoint is hit, however there is no way to control a variable continuously based on how far into a step the user is. 

**Describe the solution you'd like**
The component should expose a `progress` parameter that utilizes scrollama's `onStepProgress` option.

### PR DESCRIPTION
**Pull request by RakeshUP**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Enhancement for scroller component

* **What is the current behavior?** (You can also link to an open issue here)
There is only step callback in scroller now

* **What is the new behavior (if this is a feature change)?**
Scroller component fires a step progress event now

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
Resolves #589

### CODE REVIEW

**Comment by RakeshUP** [comment]
@mathisonian, I came up with the above solution. Should I change the docs for Scroller component too?

File: packages/idyll-components/src/scroller.js [hunk 1]
```diff
@@ -92,6 +94,12 @@ class Scroller extends React.Component {
     }
   }
 
+  handleStepProgress(response) {
+    const progress = { response };
+    const update = { progress: progress * 100 };
```
Comment (mathisonian) [inline packages/idyll-components/src/scroller.js]: At this point the value of `progress` will be `{ response: valueOfTheResponse }`, and so multiplying by 100 will lead to `NaN`.

Currently we have

```js
const progress = { response }
```

which is shorthand for:

```js
const progress = { response: response }
```

but it should instead be
```js
const  { progress } = response

// this is equivalent to:
const progress = response.progress
```

Comment (mathisonian) [inline packages/idyll-components/src/scroller.js]: Just a convention thing, but I would not multiply by 100, so the scroll progress is always between `[0, 1]`.

File: packages/idyll-components/src/scroller.js [hunk 2]
```diff
@@ -52,9 +52,11 @@ class Scroller extends React.Component {
       .setup({
         step: `#idyll-scroll-${this.id} .idyll-step`, // required
         container: `#idyll-scroll-${this.id}`, // required (for sticky)
-        graphic: `#idyll-scroll-${this.id} .idyll-scroll-graphic` // required (for sticky)
+        graphic: `#idyll-scroll-${this.id} .idyll-scroll-graphic`, // required (for sticky)
+        progress: true
```
Comment (mathisonian) [inline packages/idyll-components/src/scroller.js]: What is the performance implication of this if the user doesn't care about tracking scroll progress? Should we only set `progress` to true if they've bound a variable to the `progress` property?

File: packages/idyll-components/src/scroller.js (hunk 1 above)
Comment (RakeshUP) [reply packages/idyll-components/src/scroller.js]: I didn't check it properly, my bad. Fixed it now!

File: packages/idyll-components/src/scroller.js [hunk 3]
```diff
@@ -52,9 +52,11 @@ class Scroller extends React.Component {
       .setup({
         step: `#idyll-scroll-${this.id} .idyll-step`, // required
         container: `#idyll-scroll-${this.id}`, // required (for sticky)
-        graphic: `#idyll-scroll-${this.id} .idyll-scroll-graphic` // required (for sticky)
+        graphic: `#idyll-scroll-${this.id} .idyll-scroll-graphic`, // required (for sticky)
+        progress: this.props.progress != undefined ? true : false
```
Comment (mathisonian) [inline packages/idyll-components/src/scroller.js]: Small thing, but the project style is to always prefer `!==` over `!=` unless there's a specific reason why it isn't appropriate.

File: packages/idyll-components/src/scroller.js [hunk 4]
```diff
@@ -204,6 +212,13 @@ Scroller._idyll = {
       example: '`{}`',
       description:
         'The state value associated with the currently selected step. Note you must set the state property on the step components for this value to update.'
+    },
+    {
+      name: 'progress',
+      type: 'integer',
```
Comment (mathisonian) [inline packages/idyll-components/src/scroller.js]: `number` instead of `integer`

File: packages/idyll-components/src/scroller.js (hunk 3 above)
Comment (RakeshUP) [reply packages/idyll-components/src/scroller.js]: Sure. Thanks for pointing out

**Comment by mathisonian** [comment]
🎉

**Pull request merged**